src/flint/fmpz.jl

```diff
@@ -1165,6 +1165,22 @@ function _factor(a::fmpz)
    return res, canonical_unit(a)
 end
 
+function factor(a::Union{Int, UInt})
+   if iszero(a)
+      throw(ArgumentError("Argument is not non-zero"))
+   end
+   u = sign(a)
+   a = u < 0 ? -a : a
+   F = n_factor()
+   ccall((:n_factor, libflint), Nothing, (Ref{n_factor}, UInt), F, a)
+   res = Dict{Int, Int}()
```

Comment:
To be pedantic, shouldn't the keys have the same type as `a`? So, maybe `res = Dict{typeof(a), Int}()` here?

Reply:
This is more than pedantry, the current version doesn't work across the UInt range:
```
julia> factor(next_prime(3*UInt(2)^62))
ERROR: InexactError: check_top_bit(Int64, 13835058055282163729)
```

Reply:
Should be fixed now.